### PR TITLE
Actually print out if xcrun metal fails

### PR DESCRIPTION
**Pull request by dnfield**
My previous commit was exiting with an unhelpful error that did not contain the failure output. This prints the failure output more nicely.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
RSLGTM

File: impeller/tools/build_metal_library.py [hunk 1]
```diff
@@ -111,10 +111,17 @@ def main():
 
   command += args.source
 
-  subprocess.check_output(command, stderr=subprocess.STDOUT)
+  try:
+    subprocess.check_output(command, stderr=subprocess.STDOUT, text=True)
```
Comment (zanderso) [inline impeller/tools/build_metal_library.py:115]: The presubmits will tell, but hopefully CI has a recent enough python for `text=True`.

**Review by zanderso — APPROVED** [review]

File: impeller/tools/build_metal_library.py (hunk 1 above)
Comment (dnfield) [reply impeller/tools/build_metal_library.py:115]: I'm _pretty_ sure we do. I was going to use subprocess.run here but I'm not sure we have a recent enough for that. If it fails I'll change this.

Comment (dnfield) [reply impeller/tools/build_metal_library.py:115]:
```
impeller/tools/build_metal_library.py:116:2: C0103: Variable name "e" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern (invalid-name)
```

For real?

**Pull request merged**